### DISCUSSION TITLE
Fonts are not installed or registered with msi created by Wix 4 Toolset

### DISCUSSION THREAD

**Discussion by kerrywicks** [wixtoolset/issues#7589] [Questions] [answered]
I'm not sure if this is a bug, or if I am referencing a StandardDirectory incorrectly. I used the Wix conversion kit to upgrade from Wix 3 to Wix 4.

My installers are mostly working at this point - except I realised that the fonts are no longer getting installed. 

Here are some example snippets:

Component declaration:

```
<ComponentGroup Directory="FontsFolder" Id="FontsFolderFiles">
			<Component Id="NotoSansRegular.ttf" Guid="4646234F-D82C-464B-BDC3-10E14AF940ED">
				<File Id="NotoSansRegular.ttf" Source="$(var.thirdPartyResources)\Fonts\Gx\NotoSans-hinted\NotoSans-Regular.ttf" TrueType="yes" KeyPath="yes" />
			</Component>
</ComponentGroup>
```

Feature declaration:

```
<Feature Id="DominoAutomation" Title="DominoAutomation" Level="1" AllowAdvertise="no" AllowAbsent="no">
			<ComponentGroupRef Id="FontsFolderFiles" />
</Feature Id>
```

And finally, folder declaration:

`<StandardDirectory Id="FontsFolder" />`

Comparing my Wix 3.11 msi with my Wix  4.0.1 msi, the RegisterFonts action is no longer in the InstallExecute sequence:

![image](https://github.com/wixtoolset/issues/assets/48059830/3c75068d-8ace-4d42-88a3-df67edea65e4)

As a result, when the installer is run, the fonts are not being installed in the C:\Windows\Fonts directory. Is anyone able to help me out with what I'm doing wrong, or should I raise a bug for this? Something of note - while I doubt this is down to a case sensitivity issue, windows 10 has the directory as C:\Windows\Fonts as opposed to C:\windows\Fonts.

**Comment by robmen** [comment]
If the MSIs aren't different the issue isn't with WiX. Hopefully that helps narrow down your search.

**Comment by kerrywicks** [comment]
Thanks for taking the time to respond to this!

I just dug out the old msi to try a comparison (quite frankly, I should have tried that first). It looks like the RegisterFonts action is no longer being generated in the install execute sequence:

(left side is an msi generated by wix 3.11 toolset, right is generated by Wix 4.0.1

![image](https://github.com/wixtoolset/issues/assets/48059830/7b76b785-b38d-41e9-94df-398f3cee6bce)

If I use orca to manually add the RegisterFonts action, the fonts are getting installed as expected.

I've edited the top post to reflect this. Is this down to something I've done incorrectly?

**Comment by kerrywicks** [comment, marked as answer]
I've found the way around this - manually adding RegisterFonts to the InstallExecuteSequence in my Product.wxs file. If you would like, I can get this written up as a bug (unless this isn't considered a bug with Wix 4, and this is considered correct behaviour)? 

Adding this to your product.wxs file will ensure that fonts get registered:

`<InstallExecuteSequence>
    <RegisterFonts Sequence="5300"/>
</InstallExecuteSequence>`

Thanks again for your assistance!

**Reply by robmen** [reply]
That is definitely a WiX bug. Feel free to open an issue since we will need to fix that bug.

**Reply by kerrywicks** [reply]
I've opened https://github.com/wixtoolset/issues/issues/7593 - thanks again for your help!
